spotirec/spotirec.py

```diff
@@ -417,18 +416,37 @@ def check_tune_validity(tune: str):
         exit(1)
 
 
+def filter_list_duplicates(li: list) -> list:
+    """
+    Removes duplicates from a list
+    :param li: original list
+    :return: list without duplicates
+    """
+    # extract uris if elements are dicts
+    new_li = [x['uri'] if type(x) is dict else x for x in li]
+    # filter duplicates
+    new_li = sorted(list(set(new_li)), key=new_li.index)
+    for x in li:
+        if type(x) is dict and x['uri'] in new_li:
+            new_li[new_li.index(x['uri'])] = x
+        elif x in new_li:
+            new_li[new_li.index(x)] = x
+    return new_li
```

Comment:
I don't understand why it is not sufficient to simply cast to set and then to list again.